include/classes/Language/LanguageManager.class.php

```diff
@@ -260,13 +260,16 @@ function exists($code) {
 	// import language pack from specified file
 	// return imported AChecker version if it does not match with the current version
 	function import($filename, $ignore_version = false) {
-		require_once(AC_INCLUDE_PATH . 'lib/pclzip.lib.php');
-
+		
 		$import_path = AC_TEMP_DIR . 'import/';
 
-		$archive = new PclZip($filename);
-		if ($archive->extract(PCLZIP_OPT_PATH,	$import_path) == 0) {
-			exit('Error : ' . $archive->errorInfo(true));
+		$zip = new ZipArchive();
+	
+		if ($zip->open($filename) === TRUE) {
+			$zip->extractTo($import_path);
+			$zip->close();			
+		} else {
+			exit('Error : ' . $zip->open($filename));

```

Comment:
Error handling needs to be improved. For example, if you rename a non-zip file to a "zip" extension and upload it in Updater, the entire page will be blank and only show "Errro:19". A more human readable error should be reported using global `$msg` utility.

Reply:
When using that function, this is the result, there is still an option to continue
![errorachecker](https://user-images.githubusercontent.com/15857679/40930034-7217b816-681e-11e8-8075-1fa578495e4b.png)

Reply:
Though, it will noy add any new language

Reply:
This option is the next step after the language pack is unzipped. If the zip file fails at unzipping, the error should be displayed at the previous "Language" index page instead of this step. You need to stop the script from proceeding after calling `$msg->addError()`.